internal/app/app.go

```diff
@@ -178,6 +165,28 @@ func NewDefaultRelayer(
 	return relayer, nil
 }
 
+func NewDefaultStorage(cfg config.NeutronQueryRelayerConfig, logger *zap.Logger) (relay.Storage, error) {
+	var (
+		err            error
+		relayerStorage relay.Storage
+	)
+
+	if cfg.AllowTxQueries && cfg.StoragePath == "" {
```

Comment:
I have a feeling, `NewDefaultStorage` should know nothing about `AllowTxQueries`.

Reply:
Yeah, although it's copypaste to allow e move storage to main.go, we can do something about it

Reply:
I think it's a bit out of scope. I think current solution is good enough, but we can move it out of here later.